packages/now-build-utils/src/detect-routes.ts

```diff
@@ -337,6 +337,18 @@ export function detectApiDirectory(builders: Builder[]): string | null {
   return found ? 'api' : null;
 }
 
+export function detectApiExtensions(builders: Builder[]): Set<string> {
+  return new Set<string>(
+    builders
+      .filter(
+        b =>
+          b.config && b.config.zeroConfig && b.src && b.src.startsWith('api/')
+      )
+      .map(b => extname(b.src))
```

Comment:
It's possible that this won't work, e.g if someone uses a file without extension but specifies a function in now.json

Reply:
That is actually expected behavior because if a file doesn’t have an extension, then we won’t add a route redirect and we don’t need to rename the file. I think I added a test for that case. Maybe I should add a now dev integration test too.